### PR TITLE
Parallelize some tests

### PR DESCRIPTION
**Pull request by abadams**
This reduces the time taken to run all correctness tests from 8:15 to 3:15 on my machine.

It composes reasonably well with test sharding, but it's a bit weird that both mechanisms exist.

simd_op_check needed some reworking to avoid shared state between the threads.

### CODE REVIEW

**Comment by abadams** [comment]
This failure puzzles me: https://buildbot.halide-lang.org/master/#/builders/74/builds/59
The asan variant of simd_op_check_wasm is crashing, with what looks like an internal ASAN error. 

This is with a thread-pool of size 1, so it's not due to race conditions or anything. 

Turning off our stack switching for lowering with by setting HL_COMPILER_STACK_SIZE=0 doesn't heal it either, so it's not about that.

This crash only happens using the cmake preset. If I just use gcc, clang++-15, or the same clang++-19 as the preset with the makefile build, adding -fsanitize=address everywhere, no errors.

valgrind runs the test with no complaints

Cargo culting from online, Setting ASAN_OPTIONS=use_sigaltstack=0 *does* heal it, but I don't know why.

@steven-johnson have you seen this before? Any idea what's going on?

**Comment by steven-johnson** [comment]
No, unfortunately. ASAN can be weird sometimes. I could try pulling this into google3 and testing there.

File: test/correctness/float16_t_neon_op_check.cpp [hunk 1]
```diff
@@ -64,20 +64,20 @@ class SimdOpCheck : public SimdOpCheckTest {
...
             const int bits;
-            ImageParam in_f;
+            std::function<Expr(Expr)> in_f;
             std::vector<std::pair<int, string>> vl_params;
             Expr f_1, f_2, f_3, u_1, i_1;
         };
         // clang-format off
         TestParams test_params[2] = {
-            {32, in_f32, {{1, "s"}, {2, ".2s"}, {4, ".4s"}, { 8, ".4s"}}, f32_1, f32_2, f32_3, u32_1, i32_1},
+            {32, in_f32, {{1, "s"}, {2, ".4s"}, {4, ".4s"}, { 8, ".4s"}}, f32_1, f32_2, f32_3, u32_1, i32_1},
```
Comment (steven-johnson) [inline test/correctness/float16_t_neon_op_check.cpp:73]: What's with this change?

**Review by steven-johnson — APPROVED** [review]

File: test/correctness/float16_t_neon_op_check.cpp (hunk 1 above)
Comment (abadams) [reply test/correctness/float16_t_neon_op_check.cpp:73]: Accidental commit. Will revert.

**Pull request merged**